oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -822,7 +846,33 @@ int mca_spml_ucx_put_nb(shmem_ctx_t ctx, void* dst_addr, size_t size, void* src_
     return ucx_status_to_oshmem_nb(status);
 }
 
+int mca_spml_ucx_put_nb_wprogress(shmem_ctx_t ctx, void* dst_addr, size_t size, void* src_addr, int dst, void **handle)
+{
+    unsigned int i;
+    void *rva;
+    ucs_status_t status;
+    spml_ucx_mkey_t *ucx_mkey;
+    mca_spml_ucx_ctx_t *ucx_ctx = (mca_spml_ucx_ctx_t *)ctx;
+
+    ucx_mkey = mca_spml_ucx_get_mkey(ctx, dst, dst_addr, &rva, &mca_spml_ucx);
+    status = ucp_put_nbi(ucx_ctx->ucp_peers[dst].ucp_conn, src_addr, size,
+                     (uint64_t)rva, ucx_mkey->rkey);
+
+    if (OPAL_LIKELY(status >= 0)) {
+        mca_spml_ucx_remote_op_posted(ucx_ctx, dst);
+    }
 
+    if (++ucx_ctx->nb_progress_cnt > mca_spml_ucx.nb_put_progress_thresh) {
+        for (i = 0; i < mca_spml_ucx.nb_ucp_worker_progress; i++) {
+            if (!ucp_worker_progress(ucx_ctx->ucp_worker)) {
+                ucx_ctx->nb_progress_cnt = 0;
+                break;
+            }
+        }
+    }
```

Comment:
what about just add this loop into mca_spml_ucx_put_nb instead of separate function?